src/NzbDrone.Core/Download/TrackedDownloads/TrackedDownloadService.cs

```diff
@@ -70,6 +75,11 @@ public TrackedDownload TrackDownload(DownloadClientDefinition downloadClient, Do
                         return null;
                     }
 
+                    if (parsedEpisodeInfo == null)
```

Comment:
I don't think it should be an else so in the small case we parsed the wrong thing the map returned nothing.  I've refactored it to make more sense nonetheless.

It is my opinion that we shouldn't even parse the download name and always use the source name but I kept it close to what the original code does.